connect-inject/container_init.go

```diff
@@ -81,6 +82,13 @@ func (h *Handler) containerInit(pod *corev1.Pod, k8sNamespace string) (corev1.Co
 		panic("No service found. This should be impossible since we default it.")
 	}
 
+	// If ACLs are enabled, the ACL token is assumed to be in the ServiceAccount that
+	// shares the name with the Service. If the name of the service does not match that
+	// of the ServiceAccount, the token will not be found leading to an opaque 403 error.
+	if data.AuthMethod != "" && data.ServiceName != pod.Spec.ServiceAccountName {
+		return corev1.Container{}, errors.New(fmt.Sprintf(`serviceAccountName "%s" does not match service name "%s"`, pod.Spec.ServiceAccountName, data.ServiceName))
```

Comment:
```suggestion
		return corev1.Container{}, fmt.Errorf("serviceAccountName %q does not match service name %q", pod.Spec.ServiceAccountName, data.ServiceName)
```

I think this can be simplified. `%q` quotes the string.